src/main/java/org/terasology/thirst/ThirstAuthoritySystem.java

```diff
@@ -42,16 +49,62 @@
  * This authority system handles drink consumption by various entities.
  */
 @RegisterSystem(value = RegisterMode.AUTHORITY)
-public class ThirstAuthoritySystem extends BaseComponentSystem {
+public class ThirstAuthoritySystem extends BaseComponentSystem{
     private static final Logger logger = LoggerFactory.getLogger(ThirstAuthoritySystem.class);
     @In
     private EntityManager entityManager;
     @In
     private InventoryManager inventoryManager;
     @In
+    private PrefabManager prefabManager;
+    @In
     private Time time;
+    @In
+    private DelayManager delayManager;
 
     private boolean destroyDrink = false;
+    private static long lastTick = 0;
+    /**
+     * The interval (in milliseconds) at which healthDecreaseAmount (thirstComponent) is applied to the component.
+     */
+    private int healthDecreaseInterval = 30000;
+    public static final String THIRST_DAMAGE_ACTION_ID = "Thirst Damage";
+
+    /**
+     * Deals a unit of thirst damage to the character.
+     */
+    @ReceiveEvent
+    public void onPeriodicActionTriggered(PeriodicActionTriggeredEvent event, EntityRef entity_unused) {
+        if (event.getActionId().equals(THIRST_DAMAGE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(ThirstComponent.class, AliveCharacterComponent.class)) {
```

Comment:
I am unable to figure out whether i have to cycle through all entities with thirst and alive component, or should reducing the health of player (entity_unused) should be sufficient

Reply:
It's probably better to reduce all entities, rather than just the player.
This means you can put the `ThristComponent` on something else, like an NPC and the functionality will still work fine.  
Being able to do things like that is usually an indication of a well designed ECS system, because the ability to quickly swap around stuff is one of the key things for an ECS.